### DISCUSSION TITLE
Variable inheritance

### DISCUSSION THREAD

**Discussion by bphd** [AppImage/AppImageKit#1360] [Q&A] [answered]
How to make inherit or set a variable in an app image?

I mean in all cases, a viable set system-wide (environment variable), set in the shell configuration, set in the shell session, set in the app image command, any other case that I forgot about

Whether the app image is run, or extracted, or extracted and run, or whatever

Like, I want my inkscape.appimage be aware that I set defaultTool=pen, for the right tool to be selected in the software when working. I tried many combination and the software is never aware of the variable

**Comment by probonopd** [comment, marked as answer]
In this regard, an application behaves just like any other application.

Remember: An AppImage is just a self-mounting disk image that executes whatever its author has put in. It doesn't unset environment variables.

**Reply by bphd** [reply]
https://github.com/AppImage/AppImageKit/issues/1154#issuecomment-2465614294

https://github.com/AppImage/AppImageKit/issues/977#issuecomment-2465602523

https://github.com/AppImage/AppImageKit/issues/1131#issuecomment-2465593797
